rlp/sedes/serializable.py

```diff
@@ -174,35 +144,59 @@ def serialize(cls, obj):
             raise ObjectSerializationError(obj=obj, sedes=cls, list_exception=e)
 
     @classmethod
-    def deserialize(cls, serial, mutable=False, **kwargs):
+    def deserialize(cls, serial, **extra_kwargs):
         try:
             values = cls._meta.sedes.deserialize(serial)
         except ListDeserializationError as e:
             raise ObjectDeserializationError(serial=serial, sedes=cls, list_exception=e)
 
-        if mutable:
-            value_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
-            return cls.create_mutable(**value_kwargs, **kwargs)
-        else:
-            immutable_args = tuple(make_immutable(arg) for arg in values)
-            value_kwargs = merge_args_to_kwargs(immutable_args, {}, cls._meta.field_names)
-            return cls.create_immutable(**value_kwargs, **kwargs)
+        args_as_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
+        return cls(**args_as_kwargs, **extra_kwargs)
 
+    def copy(self, *args, **kwargs):
+        missing_overrides = set(
+            self._meta.field_names
+        ).difference(
+            kwargs.keys()
+        ).difference(
+            self._meta.field_names[:len(args)]
+        )
+        unchanged_kwargs = {
+            key: copy.deepcopy(value)
+            for key, value
+            in self.as_dict().items()
+            if key in missing_overrides
+        }
+        combined_kwargs = dict(**unchanged_kwargs, **kwargs)
+        all_kwargs = merge_args_to_kwargs(args, combined_kwargs, self._meta.field_names)
+        return type(self)(**all_kwargs)
 
-def make_immutable(value):
-    if hasattr(value, 'as_immutable'):
-        return value.as_immutable()
-    elif not isinstance(value, (bytes, str)) and isinstance(value, collections.Sequence):
-        return tuple(make_immutable(item) for item in value)
-    else:
-        return value
+    def __copy__(self):
+        return self.copy()
+
+    def __deepcopy__(self, *args):
+        return self.copy()
 
+    _in_mutable_context = False
 
-def make_mutable(value):
-    if hasattr(value, 'as_mutable'):
-        return value.as_mutable()
-    elif not isinstance(value, (bytes, str)) and isinstance(value, collections.Sequence):
-        return list(make_mutable(item) for item in value)
+    @contextlib.contextmanager
+    def build_copy(self, *args, **kwargs):
+        c_self = self.copy(*args, **kwargs)
+        c_values = tuple(c_self)
+        c_self._in_mutable_context = True
+        try:
+            yield c_self
+        except Exception:
+            for value, attr in zip(c_values, self._meta.field_attrs):
```

Comment:
*super nitpicky*

It felt backwards to do `value, attr` instead of `attr, value`